lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/Runner.scala

```diff
@@ -232,13 +232,19 @@ class Runner(
       case None =>
         runtimeVersionManager.withEngineAndRuntime(engineVersion) {
           (engine, runtime) =>
-            val ensoLauncher = Option(System.getenv(Runner.LAUNCHER_ENV_NAME))
-            val requiresJVMRunner =
-              ensoLauncher.exists(_.equals("shell")) || runSettings.jvmMode
-            if (requiresJVMRunner) {
+            if (runSettings.jvm.isDefined) {
+              val javaHome = runSettings.jvm.get
+              val javaExec = if (javaHome.isDefined) {
```

Comment:
- I wrote this code in a way I can read it later.
- I assume there is some fancy way of using `Option.map.orElse` instead
- reviewers feel free to make such a local stylistic changes directly
- you have write access to the code in this PR